argm.c

```diff
@@ -311,17 +311,25 @@ argm_combine_universal(PG_FUNCTION_ARGS, int32 compareFunctionResultToAdvance)
 
     GET_AGG_CONTEXT(fcinfo, aggcontext);
 
-    if (PG_ARGISNULL(1))
-        PG_RETURN_POINTER(PG_GETARG_POINTER(0));
+    state0 = PG_ARGISNULL(0) ? NULL : (ArgmState *) PG_GETARG_POINTER(0);
+    state1 = PG_ARGISNULL(1) ? NULL : (ArgmState *) PG_GETARG_POINTER(1);
+
+    if (state1 == NULL) {
+        /* just pass state0 down the line */
+        if (state0 == NULL)
+            PG_RETURN_NULL();
+        else
+            PG_RETURN_POINTER(state0);
+    }
     
+    /*
+     * okay, no fast track, we need to make a new value,
+     * either by copying state1 into relevant context or by building a new one
+     */
     oldcontext = MemoryContextSwitchTo(aggcontext);
 
-    state1 = (ArgmState *) PG_GETARG_POINTER(1);
-
-    if (PG_ARGISNULL(0))
+    if (state0 == NULL)
     {
-        state0 = palloc(sizeof(ArgmState));
```

Comment:
Oh, so I guess this was a memory leak before too?  Though looks like `palloc` allocates from a context anyway and is freed all at once, so maybe not such a big deal.

Reply:
Yeah must have been a leak! This memory context may be quite long-living, so it's nice we get rid of this double-allocation.